lessons/admin.py

```diff
@@ -470,6 +477,13 @@ def unit(self, instance):
     def can_access_all(self, request):
         return request.user.has_perm('lessons.access_all_activities')
 
+    def save_form(self, request, form, change):
+        lesson_id = form.cleaned_data['lesson'].id
+        lesson = Lesson.objects.get(id=lesson_id)
+        if not lesson.can_access(request):
+            raise Exception('You can only add an activity to a lesson you own')
+        return super(ActivityAdmin, self).save_form(request, form, change)
+
```

Comment:
Why do we need this twice - once for ActivityAdmin and once for ObjectiveAdmin? Maybe extract or consolidate?

Reply:
Let's see. I could extract this into a parent class:
```
def validate_form_child(self, request, form, field_name, model):
    id = form.cleaned_data[field_name].id
    object = model.objects.get(id=id)
    if not object.can_access(request)
        raise Exception("%s field must refer to a %s you own" % (field_name, field_name))
```
and then call it like this here:
```
def save_form(self, request, form, change):
    self.validate_form_child(request, form, field_name, model)
    return super(ActivityAdmin, self).save_form(request, form, change)
```
Does that look better? 

I am going to create a parent `Filterable` class in the next PR, so I could extract this in that PR.

Reply:
Ah. Thanks for the explanation. I don't feel strongly, mostly curious.  If it makes sense for follow-up, works for me.

Reply:
I was hesitant at first, but yes, I think this is a good follow-up!